cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -924,6 +957,72 @@ impl Node<Option<cst::VariableDef>> {
         }
     }
 
+    #[cfg(feature = "tolerant-ast")]
+    fn to_principal_or_resource_constraint_tolerant_ast(
+        &self,
+        expected: ast::Var,
+    ) -> Result<PrincipalOrResource> {
```

Comment:
I don't think this amount of code duplication is acceptable. Either we refactor common parts and use them both here and in the "regular" implementation, or use conditional compilation on the latter and we don't have this function. The goal is to make it easier to spot the differences between the two implementations.

Reply:
Totally makes sense! Back in the original tolerant AST [PR](https://github.com/cedar-policy/cedar/pull/1470) there was some back and forth about this and decision was to accept a certain amount of repetition

But I think you're right that here it's a little too much, so I will refactor